somnio_starter/__brick__/{{project_name.snakeCase()}}/lib/app/router/auth_stream_scope.dart

```diff
@@ -0,0 +1,39 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:{{project_name}}/app/app.dart';
+
+class AuthStreamScope extends InheritedNotifier<AuthStream> {
+  AuthStreamScope({
+    required AppBloc appBloc,
+    required super.child,
+    super.key,
+  }) : super(notifier: AuthStream(appBloc, appBloc.stream));
+
+  static AuthStream of(BuildContext ctx) =>
+      ctx.dependOnInheritedWidgetOfExactType<AuthStreamScope>()!.notifier!;
+}
+
+class AuthStream extends ChangeNotifier {
+  AuthStream(
+    AppBloc bloc,
+    Stream<dynamic> stream,
+  ) : _bloc = bloc {
+    _subscription = stream.asBroadcastStream().listen((_) => notifyListeners());
+  }
```

Comment:
I think we don't need to pass the stream, we can get the stream directly from the bloc, also we don't need to use the .asBrasdcastStream since this is a stream already

Reply:
AuthStream is not a stream itself, is a ChangeNotifier. Maybe I should rename it to avoid confusions.

I believe you are referring to the stream from the user repository but its not scoped to be used outside the bloc. The other option is using the stream directly from the repository but in that case we would jump a layer in the architecture, because the routing belongs to the UI layer. Thats why I created a separate object for the view